lxd/storage/backend_lxd.go

```diff
@@ -81,6 +81,12 @@ func (b *lxdBackend) create(localOnly bool, op *operations.Operation) error {
 
 	// If dealing with a remote storage pool, we're done now.
 	if b.driver.Info().Remote && localOnly {
+		// Create the directory structure.
```

Comment:
@monstermunchkin for other drivers, `createStorageStructure` is called after mounting the storage driver, so that the directory structure is created on the remote storage device, this way it is still present when the storage pool is mounted.

As I'm not familiar with ceph, does it not need the storage pool mounted first (presumably only the individual volumes need mounting like LVM), however if we add another remote storage pool driver in the future this may not be appropriate. Perhaps we need something in Info() to indicate that the local setup requires the directory structure to be created?

Reply:
There's no need to mount storage pools in Ceph. Only volumes are mounted.

Reply:
@monstermunchkin ok so we need some way for the driver to inform the backendLXD that this driver needs the directory structure to be created without mounting the storage pool in a `localOnly` scenario. This would ensure we don't create it unnecessarily for remote pools where the directory structure is created on the remote storage device itself.

Reply:
Well, for CEPH we actually want `createStorageStructure` to be run on every node during creation. It's effectively the same behavior as with LVM except that we need it done on all nodes.

Reply:
@stgraber yeah, I'm thinking we need a way for the driver to communicate that `createStorageStructure` should be run in `localOnly` mode, as maybe other `Remote` drivers (the kind that may need the storage pool mounted first) may not want the `createStorageStructure` to be run in `localOnly` mode. 

I'm not sure whether `cephfs` should have this happen to it now, but it would be a change of behaviour for it.

Reply:
Right, cephfs only needs createStorageStructure to happen once as the entire pool path is mounted from remote storage.

Reply:
How about b.driver.Info().LocalStructure?